src/idl_gen_cpp.cpp

```diff
@@ -551,7 +551,8 @@ class CppGenerator : public BaseGenerator {
       if (opts_.generate_object_based_api) {
         // A convenient root unpack function.
         auto native_name =
-            NativeName(WrapInNameSpace(struct_def), &struct_def, opts_);
+            WrapInNameSpace(struct_def.defined_namespace,
+                            NativeName(Name(struct_def), &struct_def, opts_));
```

Comment:
This construction `WrapInNameSpace(ns, NativeName(Name(),...)` repeated 4 times.
It would be better to add a shortcut for it.

Reply:
Done.